griptape/drivers/vector/griptape_cloud_knowledge_base_vector_store_driver.py

```diff
@@ -0,0 +1,108 @@
+        namespace: Optional[str] = None,
+        meta: Optional[dict] = None,
+        **kwargs,
+    ) -> str:
+        raise NotImplementedError(f"{self.__class__.__name__} does not support vector upsert.")
+
+    def upsert_text_artifact(
+        self,
+        artifact: TextArtifact,
+        namespace: Optional[str] = None,
+        meta: Optional[dict] = None,
+        vector_id: Optional[str] = None,
+        **kwargs,
+    ) -> str:
+        raise NotImplementedError(f"{self.__class__.__name__} does not support text artifact upsert.")
+
+    def upsert_text(
+        self,
+        string: str,
+        vector_id: Optional[str] = None,
+        namespace: Optional[str] = None,
+        meta: Optional[dict] = None,
+        **kwargs,
+    ) -> str:
+        raise NotImplementedError(f"{self.__class__.__name__} does not support text upsert.")
+
+    def load_entry(self, vector_id: str, namespace: Optional[str] = None) -> BaseVectorStoreDriver.Entry:
+        """Retrieves a specific vector entry from the collection based on its identifier and optional namespace."""
+        raise NotImplementedError(f"{self.__class__.__name__} does not support entry loading.")
+
+    def load_entries(self, namespace: Optional[str] = None) -> list[BaseVectorStoreDriver.Entry]:
+        """Retrieves all vector entries from the collection, optionally filtering to only
+        those that match the provided namespace.
+        """
+        raise NotImplementedError(f"{self.__class__.__name__} does not support entry loading.")
+
+    def load_artifacts(self, namespace: Optional[str] = None) -> ListArtifact:
+        """Retrieves all Artifacts from the collection, optionally filtering to only
+        those that match the provided namespace.
+        """
+        raise NotImplementedError(f"{self.__class__.__name__} does not support Artifact loading.")
+
+    def query(
+        self,
+        query: str,
+        count: Optional[int] = None,
+        namespace: Optional[str] = None,
+        include_vectors: Optional[bool] = None,
+        distance_metric: Optional[str] = None,
+        # GriptapeCloudKnowledgeBaseVectorStoreDriver-specific params:
+        filter: Optional[dict] = None,
+        **kwargs,
+    ) -> list[BaseVectorStoreDriver.Entry]:
+        """Performs a search on the Knowledge Base to find vectors similar to the provided input vector,
+        optionally filtering to only those that match the provided filter(s).
+        """
+        url = urljoin(self.base_url.strip("/"), f"/api/knowledge-bases/{self.knowledge_base_id}/query")
+
+        request: dict[str, Any] = {"query": query}
+        if count is not None:
+            request["count"] = count
+        if distance_metric is not None:
+            request["distance_metric"] = distance_metric
+        if filter is not None:
+            request["filter"] = filter
+        if include_vectors is not None:
+            request["include_vectors"] = include_vectors
```

Comment:
Maybe refactor to use [this syntax](https://github.com/griptape-ai/griptape/pull/951/files#diff-bf37af9210488ef8685c42ef0c8e19a8c6fb8829e00512171f1d45ebf1eb62edR136)?